File: src/main/java/ru/itmo/blpslab1/rest/AuthController.java
package ru.itmo.blpslab1.rest;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import ru.itmo.blpslab1.rest.dto.request.AuthRequest;
import ru.itmo.blpslab1.rest.dto.request.SignUpRequest;
import ru.itmo.blpslab1.rest.dto.response.AuthResponse;
import ru.itmo.blpslab1.rest.dto.response.SignUpResponse;
import ru.itmo.blpslab1.service.AuthUserService;

import java.util.function.Function;

@RestController("/api/users")
@RequiredArgsConstructor
public class AuthController {

    private AuthUserService authUserService;

    @PostMapping("/auth")
    public AuthResponse authUser(@Valid @RequestBody AuthRequest authRequest){
        return authUserService.auth(authRequest).fold(
                left -> new AuthResponse(),
                right -> right
        );
    }

    @PostMapping("/signup")
    public SignUpResponse signUpUser(@Valid @RequestBody SignUpRequest signUpRequest){
        return null;
    }
}

File: src/main/java/ru/itmo/blpslab1/rest/AuthController.kt
package ru.itmo.blpslab1.rest;

import io.swagger.v3.oas.annotations.security.SecurityRequirement
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.*
import ru.itmo.blpslab1.rest.dto.request.AuthRequest
import ru.itmo.blpslab1.rest.dto.request.SignUpRequest
import ru.itmo.blpslab1.rest.dto.request.query.UserQuery
import ru.itmo.blpslab1.service.AuthUserService
import ru.itmo.blpslab1.utils.core.toResponse

@RestController
@RequestMapping("/api/users")
@SecurityRequirement(name = "JWT")
class AuthController(
    private var authUserService: AuthUserService
) {
    @PostMapping("/auth")
    fun authUser(
        @Valid @RequestBody authRequest: AuthRequest
    ) = authUserService.auth(authRequest).toResponse()

    @PostMapping("/signup")
    fun signUpUser(
        @Valid @RequestBody signUpRequest: SignUpRequest
    ) = authUserService.signUp(signUpRequest).toResponse()

    @PostMapping("/getAll")
    fun getAll(
        @Valid @RequestBody userQuery: UserQuery
    ) = authUserService.getAll(userQuery)
}
